### ISSUE TITLE
Stuck when idle and minimized

### ISSUE THREAD

**Issue by AlexFundorin** [SchizoDuckie/DuckieTV#1325]
**What build of DuckieTV are you using (Standlone / Chrome Extension (New Tab / Browser Action))**
202108142304 https://i.imgur.com/gh9uFzT.png

**What version of DuckieTV are you using (Stable 1.1.x / Nightly yyyymmddHHMM)**
Nightly https://i.imgur.com/0HReKEH.png

**What is your Operating System (Windows, Mac, Linux, Android)**
W10

**Describe the problem you are having and steps to reproduce if available**
After some time, while being minimized to tray, DTV becomes unresponsive in a way that I can recall its window by clicking on the icon, but can't navigate the gui itself.

It happened with a couple of previous nightlies and, maybe, with a stable version. I was installing new versions, from time to time, hoping that this bug would be fixed by someone else's report, but, it seems like I should report it myself.

**Comment by garfield69** [comment]
I don't normally run Dtv 24/7 in minimised state so I've not come across this before.
I'll try to replicate this so that I can investigate.

**Comment by AlexFundorin** [comment]
Thanks. I run it as soon as the pc is turned on. Might as well add it to autostart at this point.

**Comment by garfield69** [comment]
This could be a tough one to diagnose, as I don't know how long to leave it before symptoms appear.
For all I know, my periodic checking (I'm progressively waiting longer in-between attempts)  to see if its stuck when it isn't, might reset whatever the cause is that breaks it.
In all likely hood this will be an issue with the 3rd party NWJS that we use to run the Dtv browser extension AngularJS code under.
My dev box is running the latest NWJS 0.56 (with chromium 93), while the Nightly packages from the bot are NWJS 0.51.1 with chromium 88.

**Comment by AlexFundorin** [comment]
I installed the latest nightly today. Had about an hour long window and DTV is still working fine.

**Comment by AlexFundorin** [comment]
Got it stuck. See, the cursor is in a form of hand, instead of a regular arrow. https://i.imgur.com/CGaD5IQ.png

**Comment by garfield69** [comment]
what view from Dtv displays episodes in the way that you are seeing, I'm not familiar with it. 
Does not look like the calendar, or the todo page, or the favourites or anything else I'm familiar with... 
Does Dtv in that state respond to the **F12** key to open the _development tools_ page, and if it does, then click on the **console** _tab_ and check to log. Any signs of errors or unusual messages?

**Comment by garfield69** [comment]
Oh I see that is the TODO page, I'm just used to seeing posters with pictures, the _into the night_'s poster is just that text, which threw me off.
I'll leave my dev Dtv in **todo** mode and see if that increases the likely hood of a lockup for me.

**Comment by AlexFundorin** [comment]
It's a calendar - https://i.imgur.com/8iR6xuK.png

**Comment by garfield69** [comment]
I wonder if this is an artefact of the Dtv midnight reset (which is a daily event where Dtv performs an internal restart so that the clock for all the Dtv tasks can restart for the new day).

Try this: 
navigate **dtv>settings>window>windows on startup** and _click_ on the **Click for Minimal** button.
Also on the **System Tray** section, set **always show tray icon**, and **minimise button always minimises DuckieTV to Tray** and check those boxes.
![image](https://user-images.githubusercontent.com/6933240/132903412-0ca45385-213c-4ee7-b533-10028f2d66de.png)

This will cause Dtv on the midnight reset to auto minimise the window so that when you open it from the system tray it will have focus. And whenever you startup Dtv from scratch its also minimised so you use the system tray to open the window rather than have it open all the time.

Perhaps that will bypass the issue.

**Comment by AlexFundorin** [comment]
I feel like I already set this from the start. These are my settings:
![T0K3lI0](https://user-images.githubusercontent.com/22612802/132904949-90e1dcec-18b4-4a87-8287-8295109ae2ec.png)

**Comment by garfield69** [comment]
ok, good.
so to open the window do you just **click** on the system tray?
or do you **right click** and select the **Show Calendar** menu option?
or do you **right click** and select the **Show DuckieTV** menu option?

doe any of these alternate methods make any difference to you?

Also, if the window is open but wont regain focus, does using the System Tray right click and selecting, say for instance, Show About. Does then clicking on the Top Left DTv icon restore the calendar and give you access again?

Just exploring options....

**Comment by AlexFundorin** [comment]
I always left click on the tray icon every time. DTV is set to calendar.
The window is opening, but I'm enable to navigate, although the cursor form changes from hand to arrow when I click on the left vertical icons.

So, are you suggesting me to right click for a couple of days to see if it's still hanging, selecting different options? Will do.
I just restarted DTV to change settings, as you said in your previous comment, so we need to wait a bit more till it hangs again.

**Comment by AlexFundorin** [comment]
I clicked on Calendar in the tray context menu and showed up in a stuck mode, as usual.
But, after I closed it with an X and reopened, but choosing Settings this time, it switched to the settings layout and became responsive again.

**Comment by garfield69** [comment]
Good to see you can at least avoid having to restart Dtv, it does appear to be a window is out of focus issue of some kind.

I wonder if the problem is restricted to just the TODO page... 
Something to try for the next couple days, would be to switch the TODO back to the calendar view, and if the monthly view is too busy looking for you, set it to weekly via the **Dtv>settings>calendar>Calendar Display Mode** and _click_ on the **Use One-Week Calendar** button.

**Comment by AlexFundorin** [comment]
I'm kinda used to restarting, at this point, tbh. 😊

Never used calendar mode. Turns out that what I called calendar is todo page. Todo is what I use.
I can stick to calendar view for testing purposes. No problem.
Do you want me to call DTV by choosing Calendar from the context menu and also try other options as well?

**Comment by garfield69** [comment]
At first, start by using what ever method you usually use to open the window, and if the calendar window is unresponsive then try the system tray Dtv>Show Settings to get control back.
But then for the next time try the other ways to open the window and see if they give you control.
IF however the calendar widow appears to always work then that will be progress in narrowing down the issue.

IF we can isolate the issue to a particular set of circumstances then I've got something to work with to try to see if there is anything that can be done about it.

BTW, thank you for persevering with these tests, its appreciated.

**Comment by AlexFundorin** [comment]
Still works in calendar view, as opposed to todo. Even after waking my pc from sleep mode.

**Comment by AlexFundorin** [comment]
Wepl. Stuck again. Calendar. 

context>show dtv didn't work
context>favorites worked.

BTW, when it's stuck, I'm also unable to use min/max/close buttons.

**Comment by garfield69** [comment]
yeah, it looks like the whole window is out of focus and there is no place to click to get focus back.

ok that's enough testing/feedback, thank you.

for now, since you have a way of getting control back, I'll leave you in peace while I take a look and see if there is anything can be done about this.
I think I have a dev method I can use to kick off the midnight reset at at a set time, so I can experiment on my sandbox and see what happens without waiting overnight for results.
